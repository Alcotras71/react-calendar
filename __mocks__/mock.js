export const mockEventData = {
  eventDataArr: [
    {
      touchedDate: "2021-07-27T21:00:00.000Z",
      startTime: "12:00",
      endTime: "12:30",
      eventName: "mocki",
    },
    {
      touchedDate: "2021-09-27T21:00:00.000Z",
      startTime: "16:00",
      endTime: "18:30",
      eventName: "Loki-moki",
    },
    {
      touchedDate: "2021-07-29T21:00:00.000Z",
      startTime: "9:00",
      endTime: "10:30",
      eventName: "PockMocki",
    },
  ],
  eventData: {
    touchedDate: "2021-07-27T21:00:00.000Z",
    startTime: "12:00",
    endTime: "12:30",
    eventName: "mocki",
  },
  eventData2: {
    touchedDate: "2021-08-27T21:00:00.000Z",
    startTime: "11:00",
    endTime: "15:30",
    eventName: "mocddd",
  },
};

export const mockWeatherData = {
  coord: { lon: 37.6111, lat: 54.2044 },
  weather: [{ id: 804, main: "Clouds", description: "пасмурно", icon: "04d" }],
  base: "stations",
  main: {
    temp: 18.57,
    feels_like: 18.57,
    temp_min: 18.57,
    temp_max: 18.57,
    pressure: 1014,
    humidity: 80,
    sea_level: 1014,
    grnd_level: 996,
  },
  visibility: 10000,
  wind: { speed: 2.53, deg: 155, gust: 7.58 },
  clouds: { all: 100 },
  dt: 1627476561,
  sys: { country: "RU", sunrise: 1627436115, sunset: 1627493797 },
  timezone: 10800,
  id: 480562,
  name: "Тула",
  cod: 200,
};
