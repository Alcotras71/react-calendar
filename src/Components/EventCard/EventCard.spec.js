import "jsdom-global/register";
import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import ConnectedEventCard, { EventCard } from "./EventCardContainer";

import { mockEventData } from "../../../__mocks__/mock";
import {
  getEventData,
  GET_EVENT_DATA,
  toggleCard,
  TOGGLE_CARD,
} from "../../redux/datepickerReducer";

describe("EventCard component", () => {
  const setUp = (props) => shallow(<EventCard {...props} />);
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("render EventCard with props", () => {
    component = setUp({
      eventData: mockEventData.eventDataArr,
      touchedDate: mockEventData.eventData.touchedDate,
      isOpen: true,
      cardType: "eventList",
    });
    expect(component).toMatchSnapshot();
  });

  it("render EventCard without props", () => {
    expect(component).toMatchSnapshot();
  });

  it("render the DUMB component", () => {
    expect(component.length).toEqual(1);
  });
});

describe("Connected EventCard component", () => {
  const initialState = {
    datepicker: {
      eventData: mockEventData.eventDataArr,
      cardType: "eventList",
      touchedDate: mockEventData.eventData.touchedDate,
      isOpen: false,
    },
  };
  const mockStore = configureStore();
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <ConnectedEventCard />
      </Provider>
    );
  });

  it("+++ render the connected(SMART) component", () => {
    expect(wrapper.find(ConnectedEventCard).length).toEqual(1);
  });

  it("+++ check Prop matches with initialState", () => {
    expect(wrapper.find(EventCard).prop("isOpen")).toEqual(
      initialState.datepicker.isOpen
    );
  });

  it("+++ check action on dispatching ", () => {
    let action;
    store.dispatch(toggleCard({ isOpen: true, cardType: "eventList" }));
    store.dispatch(getEventData(mockEventData.eventData2));
    action = store.getActions();
    expect(action[0].type).toBe(TOGGLE_CARD);
    expect(action[1].type).toBe(GET_EVENT_DATA);
  });
});
