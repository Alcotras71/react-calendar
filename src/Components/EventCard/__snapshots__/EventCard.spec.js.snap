// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard component render EventCard with props 1`] = `
<div
  className="event-card active"
>
  <div
    className="event-card__form-wrapper"
  >
    <EventList
      pickedData={
        Array [
          Object {
            "endTime": "12:30",
            "eventName": "mocki",
            "startTime": "12:00",
            "touchedDate": "2021-07-27T21:00:00.000Z",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`EventCard component render EventCard without props 1`] = `
<div
  className="event-card "
>
  <div
    className="event-card__form-wrapper"
  >
    <EventList />
  </div>
</div>
`;
